pkg/controller/apicurioregistry/context.go

```diff
@@ -1,132 +0,0 @@
-package apicurioregistry
-
-import (
-	ar "github.com/Apicurio/apicurio-registry-operator/pkg/apis/apicur/v1alpha1"
-	"github.com/go-logr/logr"
-	"k8s.io/apimachinery/pkg/runtime"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller"
-)
-
-// A long-lived singleton container for shared components
-type Context struct {
-	// More static stuff
-	scheme       *runtime.Scheme
-	log          logr.Logger
-	nativeClient client.Client
-	c            controller.Controller
-
-	controlFunctions []ControlFunction
-	// Components
-	configuration *Configuration
-	kubeFactory   *KubeFactory
-	ocpFactory    *OCPFactory
-	patchers      *Patchers
-	clients       *Clients
-
-	// ===
-	resourceCache ResourceCache
-	envCache      EnvCache
-
-	// ===
-	requeue bool
-}
-
-// Create a new context when the operator is deployed, provide mostly static data
-func NewContext(c controller.Controller, scheme *runtime.Scheme, log logr.Logger, client client.Client) *Context {
```

Comment:
I just noticed, the `services     map[string]interface{}` of singelton services was introduced with this PR, don't merge this, we should improve this before merging